plugins/techdocs/src/reader/transformers/sanitizeDOM.ts

```diff
@@ -34,14 +34,51 @@ export const safeLinksHook = (node: Element) => {
   return node;
 };
 
+const filterIframeHook = (allowedIframeHosts: string[]) => (node: Element) => {
+  if (node.nodeName === 'IFRAME') {
+    const src = node.getAttribute('src');
+    if (!src) {
+      node.remove();
+      return node;
+    }
+
+    try {
+      const srcUrl = new URL(src);
+      const isMatch = allowedIframeHosts.some(host => srcUrl.host === host);
+      if (!isMatch) {
+        node.remove();
+      }
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.warn(`Invalid iframe src, ${error}`);
+      node.remove();
+    }
+  }
+  return node;
+};
+
+import { Config } from '@backstage/config';
 import DOMPurify from 'dompurify';
 import type { Transformer } from './transformer';
 
-export const sanitizeDOM = (): Transformer => {
+export const sanitizeDOM = (config?: Config): Transformer => {
+  const allowedIframeHosts =
+    config?.getOptionalStringArray('allowedIframeHosts') || [];
+
   return dom => {
     DOMPurify.addHook('afterSanitizeAttributes', safeLinksHook);
+    const addTags = ['link'];
+
+    if (allowedIframeHosts.length > 0) {
+      DOMPurify.addHook(
+        'beforeSanitizeElements',
+        filterIframeHook(allowedIframeHosts),
+      );
+      addTags.push('iframe');
```

Comment:
Is there any default sanitation that's applied to the iframe elements? It's very important that we filter out vulnerable attributes like `onload`. If it isn't already done it would be preferable if it's done with a 3rd party library, explicit allow list, or based on solid and referenced security recommendations.

Reply:
Alright, good. Could you add some tests for those cases as well? Another important one is `srcdoc`, so we wanna make sure that a valid `src` does not allow an element with a `srcdoc` through for example

Reply:
@Rugvip test added :)